### ISSUE TITLE
v1.20.1-3.1.0.0 completely breaks Serene Seasons server-side

### ISSUE THREAD

**Issue by NovaViper** [wendall911/BetterDays#2]
I updated my server to the latest build but the update actually completely breaks Serene Seasons, which makes the server crash instantly when it loads the mod. Teverting back to the last version fixes the issue

The error it shows:
```
[16Aug2023 10:13:21.424] [modloading-worker-0/WARN] [mixin/]: Error loading class: net/minecraft/client/Minecraft (java.lang.RuntimeException: Attempted to load class net/minecraft/client/Minecraft for invalid dist DEDICATED_SERVER)
[16Aug2023 10:13:21.424] [modloading-worker-0/ERROR] [net.minecraftforge.fml.javafmlmod.FMLModContainer/LOADING]: Failed to create mod instance. ModID: sereneseasons, class sereneseasons.core.SereneSeasons
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?]
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:68) ~[javafmllanguage-1.20.1-47.1.44.jar%23409!/:?]
	at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156) ~[modlauncher-10.0.9.jar%2355!/:10.0.9+10.0.9+main.dcd20f30]
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135) ~[securejarhandler-2.1.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at sereneseasons.init.ModHandlers.<clinit>(ModHandlers.java:14) ~[SereneSeasons-1.20.1-9.0.0.43.jar%23378!/:9.0.0.43]
	at sereneseasons.core.SereneSeasons.<init>(SereneSeasons.java:49) ~[SereneSeasons-1.20.1-9.0.0.43.jar%23378!/:9.0.0.43]
	... 14 more
Caused by: org.spongepowered.asm.mixin.throwables.ClassMetadataNotFoundException: net.minecraft.client.Minecraft
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.transformMethod(MixinPreProcessorStandard.java:754) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.transform(MixinPreProcessorStandard.java:739) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attach(MixinPreProcessorStandard.java:310) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.createContextFor(MixinPreProcessorStandard.java:280) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinInfo.createContextFor(MixinInfo.java:1288) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:292) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:383) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:365) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:250) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.service.modlauncher.MixinTransformationHandler.processClass(MixinTransformationHandler.java:131) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.processClass(MixinLaunchPluginLegacy.java:131) ~[mixin-0.8.5.jar%2365!/:0.8.5+Jenkins-b310.git-155314e6e91465dad727e621a569906a410cd6f4]
	at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClassWithFlags(ILaunchPluginService.java:156) ~[modlauncher-10.0.9.jar%2355!/:10.0.9+10.0.9+main.dcd20f30]
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135) ~[securejarhandler-2.1.10.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at sereneseasons.init.ModHandlers.<clinit>(ModHandlers.java:14) ~[SereneSeasons-1.20.1-9.0.0.43.jar%23378!/:9.0.0.43]
	at sereneseasons.core.SereneSeasons.<init>(SereneSeasons.java:49) ~[SereneSeasons-1.20.1-9.0.0.43.jar%23378!/:9.0.0.43]
	... 14 more
[16Aug2023 10:13:21.448] [modloading-worker-0/INFO] [net.minecraftforge.common.ForgeMod/FORGEMOD]: Forge mod loading, version 47.1.44, for MC 1.20.1 with MCP 20230612.114412
[16Aug2023 10:13:21.448] [modloading-worker-0/INFO] [net.minecraftforge.common.MinecraftForge/FORGE]: MinecraftForge v47.1.44 Initialized
[16Aug2023 10:13:21.513] [modloading-worker-0/INFO] [NoChatReports/]: KONNICHIWA ZA WARUDO!
[16Aug2023 10:13:21.513] [modloading-worker-0/INFO] [NoChatReports/]: Default JVM text encoding is: UTF-8
[16Aug2023 10:13:21.518] [modloading-worker-0/INFO] [NoChatReports/]: Reading config file NoChatReports/NCR-Common.json...
[16Aug2023 10:13:21.518] [modloading-worker-0/INFO] [NoChatReports/]: Writing config file NoChatReports/NCR-Common.json...
[16Aug2023 10:13:21.524] [modloading-worker-0/INFO] [Animal Feeding Trough/]: [Animal Feeding Trough] Load Complete! Enjoy :D
[16Aug2023 10:13:21.541] [modloading-worker-0/INFO] [Puzzles Lib/]: Constructing common components for mindfuldarkness:main
[16Aug2023 10:13:21.625] [modloading-worker-0/INFO] [Puzzles Lib/]: Constructing common components for diagonalfences:main
[16Aug2023 10:13:21.633] [modloading-worker-0/INFO] [snownee.kiwi.Kiwi/INIT]: Processing 3 KiwiModule annotations
[16Aug2023 10:13:21.634] [modloading-worker-0/INFO] [Trample No More/]: Loaded config file.
[16Aug2023 10:13:21.634] [modloading-worker-0/INFO] [Trample No More/]: Saved config file.
[16Aug2023 10:13:21.648] [modloading-worker-0/WARN] [Visuality/]: Visuality is installed on a dedicated server, skip loading...
[16Aug2023 10:13:21.650] [modloading-worker-0/INFO] [Puzzles Lib/]: Constructing common components for puzzleslib:main
[16Aug2023 10:13:21.672] [modloading-worker-0/INFO] [fr.samlegamer.macawsbridgesbop.MacawsBridgesBOP/]: Macaw's Bridges - Biome O' Plenty : Loading ...
[16Aug2023 10:13:21.688] [modloading-worker-0/INFO] [fr.samlegamer.macawsbridgesbop.MacawsBridgesBOP/]: Macaw's Bridges - Biome O' Plenty : Is Charged
[16Aug2023 10:13:21.689] [modloading-worker-0/INFO] [fr.samlegamer.mcwfencesbop.McwFencesBOP/]: Macaw's Fences - Biome O' Plenty : Loading ...
[16Aug2023 10:13:21.705] [modloading-worker-0/INFO] [fr.samlegamer.mcwfencesbop.McwFencesBOP/]: Macaw's Fences - Biome O' Plenty : Is Charged
[16Aug2023 10:13:21.887] [modloading-worker-0/INFO] [thedarkcolour.kotlinforforge.test.KotlinForForge/]: Kotlin For Forge Enabled!
[16Aug2023 10:13:22.481] [main/FATAL] [net.minecraftforge.fml.ModLoader/LOADING]: Failed to complete lifecycle event CONSTRUCT, 1 errors found
[16Aug2023 10:13:22.734] [main/FATAL] [net.minecraftforge.common.ForgeMod/]: Preparing crash report with UUID 6c11ee34-b1dd-4e92-9038-4d33d34e332b
[16Aug2023 10:13:22.735] [main/FATAL] [net.minecraftforge.server.loading.ServerModLoader/]: Crash report saved to ./crash-reports/crash-2023-08-16_10.13.22-fml.txt
[16Aug2023 10:13:22.738] [main/FATAL] [net.minecraftforge.common.ForgeMod/]: Preparing crash report with UUID bf09b88b-1006-4b02-b1e9-426069d0fba6
[16Aug2023 10:13:22.739] [main/ERROR] [net.minecraft.server.Main/FATAL]: Failed to start the minecraft server
net.minecraftforge.fml.LoadingFailedException: Loading errors encountered: [
	Serene Seasons (sereneseasons) has failed to load correctly
§7java.lang.reflect.InvocationTargetException: null
]
	at net.minecraftforge.fml.ModLoader.waitForTransition(ModLoader.java:246) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at net.minecraftforge.fml.ModLoader.lambda$dispatchAndHandleError$20(ModLoader.java:210) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:210) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$13(ModLoader.java:183) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:183) ~[fmlcore-1.20.1-47.1.44.jar%23408!/:?]
	at net.minecraftforge.server.loading.ServerModLoader.load(ServerModLoader.java:30) ~[forge-1.20.1-47.1.44-universal.jar%23412!/:?]
	at net.minecraft.server.Main.main(Main.java:125) ~[server-1.20.1-20230612.114412-srg.jar%23407!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111) ~[fmlloader-1.20.1-47.1.44.jar%2369!/:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.serverService(CommonLaunchHandler.java:103) ~[fmlloader-1.20.1-47.1.44.jar%2369!/:?]
	at net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$makeService$0(CommonServerLaunchHandler.java:27) ~[fmlloader-1.20.1-47.1.44.jar%2369!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar%2355!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
```

System information:
```
Java version: 17.0.1
Server Forge: 47.1.44
Minecraft version: 1.20.1
Serene Seasons Version: 9.0.0.43
```

**Comment by NovaViper** [comment]
AH I just found the issue. It wasn't this mod actually, but it was the [Serene Seasons Fix mod ](https://www.curseforge.com/minecraft/mc-mods/serene-seasons-fix). The fix that this mod implements directly seems to conflict with the changes that the fix mod implements

**Comment by NovaViper** [comment]
Scratch that, the build is still crashing even without serene seasons fix installed

**Comment by wendall911** [comment]
Ok, let me do some testing. I need to check a couple things. I also need to block the other mod.

**Comment by wendall911** [comment]
Ok, it is loading and working well for me. I'm using: Forge 47.1.1, Serene Seasons 9.0.0.43, Better Day 3.1.0.0 (From CurseForge)

**Comment by wendall911** [comment]
Yes, that other mod does bad things, as it uses Override for a method, it should just inject. I'm not clear why it would crash still with that mod removed. Please provide a crash report for what is happening, as I can't reproduce.

**Comment by wendall911** [comment]
You can also disable the feature via the config and continue using the other fix mod.

**Comment by wendall911** [comment]
Actually, I have it set to automatically disable if the other mod is detected, so it won't even apply the mixin capabilities. I need to build the other mod locally and see if it is working correctly. I'm not entirely sure why you would get a conflict, as there isn't anything to conflict at this point.

**Comment by wendall911** [comment]
Ok, I've built sereneseasonsfix for 1.20.1 and add it. The mixin is automatically disabled. Not sure why you're having an issue. I don't see anything about a failure in betterdays. Maybe there is an error farther back in the log? I have no idea at this point, as I cannot reproduce this. Literally the mixin is disabled if the other mod is loaded.

**Comment by wendall911** [comment]
Ok, I reproduced and am checking on a fix. Thanks for the report!

**Comment by wendall911** [comment]
Ok, fixed and released. It will work as expected if you are still wanting to use sereneseasonsfix. I think my implementation is better, as it also fixes some issues with sereneseasons when using /time set and /season set

Feel free to test both. Apologies for not testing this on a dedicated server before releasing. I typically do. Heat locally is getting to me, lol

**Comment by wendall911** [comment]
Will take a few minutes to be available on CurseForge. Will be on Modrinth now.

**Comment by NovaViper** [comment]
Thank you for the patch! I just installed it onto the server and it's working! I actually gonna leave the sereneseasonsfix mod since betterdays also includes the same fixes that the mod addresses

**Comment by NovaViper** [comment]
Huh.. uh @wendall911 I'm noticing now the season's keep switching between Early Spring and Early Winter very rapidly. I don't have the other mod added but I'm suspecting it's the patch that's causing this issue (the issue wasn't present before the update also). But I need to test the old build again in order to confirm if it is indeed the patch again

**Comment by wendall911** [comment]
Ok, it may be an issue with this update. There isn't anything other than the inclusion of the serene seasons stuff in this build, so reverting back to the old one until I can figure out why servers aren't working would be best for now.

Also, try just starting the server without betterdays. Stop server. Add it back and see if the season skipping still happens.

**Comment by NovaViper** [comment]
Going back to the previous version as well as removing it from the server made the seasons go back to normal and not skip. It's definitely something with the patch causing issues with the server

**Comment by wendall911** [comment]
Ok, thanks for investigating. I'll do a bunch more testing with using this server-side. I'll update this issue once it is resolved.

**Comment by wendall911** [comment]
Ok, fixed in 3.1.0.2. If using the sereneseasonsfix, this should do nothing. Either fix works.
